tofhir-server/src/main/scala/io/tofhir/server/endpoint/ProjectEndpoint.scala

```diff
@@ -45,7 +45,7 @@ class ProjectEndpoint(schemaRepository: ISchemaRepository,
             val projectExists: Future[Option[Project]] = service.getProject(projectId)
             onSuccess(projectExists) {
               case None => complete {
-                StatusCodes.NotFound -> s"Project with id $projectId not found"
+                StatusCodes.NotFound -> (ResourceNotFound("Project not found", s"Project with id $projectId not found").toString)
```

Comment:
Instead of stringifying ResourceNotFound error, we can directly throw it as follows:
```suggestion
                throw ResourceNotFound("Project not found", s"Project with id $projectId not found")
```